TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php

```diff
@@ -118,6 +118,10 @@ public function getContextPath()
      */
     public function setName($newName)
     {
+        if ($this->getName() === $newName) {
+            return;
+        }
+
```

Comment:
This was moved up from line 129 as it is the fastest check and it actually doesn't matter if newName follows the match pattern when it is already the name of the Node.